source/diagnostics.tex

```diff
@@ -530,158 +530,83 @@
-#define ENOEXEC @\seebelow@                                              // freestanding
-#define ENOLCK @\seebelow@                                               // freestanding
-#define ENOLINK @\seebelow@                                              // freestanding
-#define ENOMEM @\seebelow@                                               // freestanding
-#define ENOMSG @\seebelow@                                               // freestanding
-#define ENOPROTOOPT @\seebelow@                                          // freestanding
-#define ENOSPC @\seebelow@                                               // freestanding
-#define ENOSYS @\seebelow@                                               // freestanding
-#define ENOTCONN @\seebelow@                                             // freestanding
-#define ENOTDIR @\seebelow@                                              // freestanding
-#define ENOTEMPTY @\seebelow@                                            // freestanding
-#define ENOTRECOVERABLE @\seebelow@                                      // freestanding
-#define ENOTSOCK @\seebelow@                                             // freestanding
-#define ENOTSUP @\seebelow@                                              // freestanding
-#define ENOTTY @\seebelow@                                               // freestanding
-#define ENXIO @\seebelow@                                                // freestanding
-#define EOPNOTSUPP @\seebelow@                                           // freestanding
-#define EOVERFLOW @\seebelow@                                            // freestanding
-#define EOWNERDEAD @\seebelow@                                           // freestanding
-#define EPERM @\seebelow@                                                // freestanding
-#define EPIPE @\seebelow@                                                // freestanding
-#define EPROTO @\seebelow@                                               // freestanding
-#define EPROTONOSUPPORT @\seebelow@                                      // freestanding
-#define EPROTOTYPE @\seebelow@                                           // freestanding
-#define ERANGE @\seebelow@                                               // freestanding
-#define EROFS @\seebelow@                                                // freestanding
-#define ESPIPE @\seebelow@                                               // freestanding
-#define ESRCH @\seebelow@                                                // freestanding
-#define ETIMEDOUT @\seebelow@                                            // freestanding
-#define ETXTBSY @\seebelow@                                              // freestanding
-#define EWOULDBLOCK @\seebelow@                                          // freestanding
-#define EXDEV @\seebelow@                                                // freestanding
+#define @\libglobal{errno}@ @\seebelow@
```

Comment:
I think I want a new LaTeX macro "\libmacro" that is currently defined as "\libglobal", but could be defined to something else.  For example, if we ever want to have a separate index of macro names or color macro definitions differently.

Reply:
Added both \libmacro and \libmacrox.  They work for both object-like and function-like macros, but I have not managed so solve the issue with macro names using placeholders.

With this change, we are close to being able to create an index of macros, that would also want the core language predefined macros to be included too.